File: packages/go-sdk/operations/deployments.go
package operations

import "context"

// DeploymentsService provides deployment management operations.
type DeploymentsService struct {
	r Requester
}

// NewDeploymentsService creates a new DeploymentsService.
func NewDeploymentsService(r Requester) *DeploymentsService {
	return &DeploymentsService{r: r}
}

// List lists deployment versions.
func (s *DeploymentsService) List(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/deployments", query, nil, nil, &result)
	return result, err
}

// Create creates a deployment version.
func (s *DeploymentsService) Create(ctx context.Context, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", "/v1/deployments", nil, nil, body, &result)
	return result, err
}

// Finalize finalizes a deployment version.
func (s *DeploymentsService) Finalize(ctx context.Context, deploymentID string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/deployments/{deploymentID}/finalize", map[string]string{"deploymentID": deploymentID}), nil, nil, body, &result)
	return result, err
}

// Promote promotes a deployment version.
func (s *DeploymentsService) Promote(ctx context.Context, deploymentID string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/deployments/{deploymentID}/promote", map[string]string{"deploymentID": deploymentID}), nil, nil, body, &result)
	return result, err
}

// Rollback rolls back to a previous deployment version.
func (s *DeploymentsService) Rollback(ctx context.Context, deploymentID string, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/deployments/{deploymentID}/rollback", map[string]string{"deploymentID": deploymentID}), nil, nil, body, &result)
	return result, err
}

File: packages/rust-sdk/src/operations/deployments.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use crate::http::substitute_path_params;
use std::sync::Arc;

pub struct DeploymentsService {
    client: Arc<StraitClient>,
}

impl DeploymentsService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/deployments", query, None, None).await
    }

    pub async fn create(&self, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("POST", "/v1/deployments", None, None, Some(body)).await
    }

    pub async fn finalize(&self, deployment_id: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/deployments/{deploymentID}/finalize", &[("deploymentID", deployment_id)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }

    pub async fn promote(&self, deployment_id: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/deployments/{deploymentID}/promote", &[("deploymentID", deployment_id)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }

    pub async fn rollback(&self, deployment_id: &str, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/deployments/{deploymentID}/rollback", &[("deploymentID", deployment_id)]);
        self.client.do_request("POST", &path, None, None, Some(body)).await
    }
}
